pkgs/development/python-modules/home-assistant-chip-wheels/default.nix

```diff
@@ -0,0 +1,305 @@
+
+      # compress to not exceed the 2GB output limit
+      # try to make a deterministic tarball
+      tar -I 'gzip -n' \
+      --sort=name \
+      --mtime=1970-01-01 \
+      --owner=0 --group=0 \
+      --numeric-owner --mode=go=rX,u+rw,a-s \
+      --transform='s@^@source/@S' \
+      -cf temp  -C "$out" .
+      rm -r "$out"
+      mv temp "$out"
+    '';
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    gn1924
+    pkg-config
+    ninja
+    clang-tools
+    zap-chip
+    # gdbus-codegen
+    glib
+    python3
+    # dependencies of build scripts
+    click
+    jinja2
+    lark
+    setuptools
+    stringcase
+    build
+    pip-tools
+    black
+    yapf
+  ];
+
+  buildInputs = [
+    glib
+    libnl
+  ];
+
+  postPatch = ''
+    cd connectedhomeip
+    export HOME=$(mktemp -d)
+
+    patchShebangs --build scripts
+
+    for patch in ../*.patch; do
+      patch -p1 < $patch
+    done
+
+    # unpin dependencies
+    # there are many files to modify, in different formats
+    sed -i 's/==.*$//' third_party/pigweed/repo/pw_env_setup/py/pw_env_setup/virtualenv_setup/python_base_requirements.txt
+    sed -i 's/==[^;]*//' scripts/setup/constraints.txt
+    sed -i 's/\(^ \+[a-zA-Z0-9-]*\)[=~><]=[^;]*/\1/' third_party/pigweed/repo/pw_protobuf_compiler/py/setup.cfg third_party/pigweed/repo/pw_protobuf/py/setup.cfg third_party/pigweed/repo/pw_protobuf_compiler/py/setup.cfg
+    # remove a few dependencies not packaged in nixpkgs and which are apparently
+    # not needed to build the python bindings of chip
+    sed -i -e '/sphinxcontrib-mermaid/d' -e '/types-six/d' -e '/types-pygment/d' -e '/types-pyserial/d' third_party/pigweed/repo/*/py/setup.cfg
+
+    # obtained by running a build in nix-shell with internet access
+    cp ${./pigweed_environment.gni} build_overrides/pigweed_environment.gni
+
+    # some code is generated by a templating tool (zap-cli)
+    scripts/codepregen.py ./zzz_pregenerated/
+  '';
+
+  # the python parts of the build system work as follows
+  # gn calls pigweed to read a dozen different files to generate
+  # a file looking like requirements.txt. It then calls pip
+  # to install this computed list of dependencies into a virtualenv.
+  # Of course, pip fails in the sandbox, because it cannot download
+  # the python packages.
+  # The documented way of doing offline builds is to create a folder
+  # with wheel files for all dependencies and point pip to it
+  # via its configuration file or environment variables.
+  # https://pigweed.dev/python_build.html#installing-offline
+  # The wheel of a python package foo is available as foo.dist.
+  # So that would be easy, but we also need wheels for transitive dependencies.
+  # the function saturateDependencies below computes this transitive closure.
+  #
+  # yes this list of dependencies contains both build tools and proper dependencies.
+  env.PIP_NO_INDEX = "1";
+  env.PIP_FIND_LINKS =
+    let
+      dependencies = [
```

Comment:
We're going to have to keep this list up-to-date, right? I'm not very familiar with nixpkgs python. Is it possible to generate this list with something like `pip2nix` to make updates easy?

Reply:
the package is not in pypi and does not follow any convention to declare dependencies so I doubt that.